src/current.rs

```diff
@@ -34,23 +40,74 @@ pub fn clock_for(task_id: TaskId) -> VectorClock {
     ExecutionState::with(|state| state.get_clock(task_id).clone())
 }
 
+/// Apply the given function to the Labels for the specified task
+pub fn with_labels_for_task<F, T>(task_id: TaskId, f: F) -> T
+where
+    F: FnOnce(&mut Labels) -> T,
+{
+    LABELS.with(|cell| {
+        let mut map = cell.borrow_mut();
+        let m = map.entry(task_id).or_default();
+        f(m)
+    })
+}
+
+/// Get a label of the given type for the specified task, if any
+#[inline]
+pub fn get_label_for_task<T: Clone + Debug + 'static>(task_id: TaskId) -> Option<T> {
+    with_labels_for_task(task_id, |labels| labels.get().cloned())
+}
+
+/// Add the given label to the specified task, returning the old label for the type, if any
+#[inline]
+pub fn set_label_for_task<T: Clone + Debug + 'static>(task_id: TaskId, value: T) -> Option<T> {
+    with_labels_for_task(task_id, |labels| labels.insert(value))
+}
+
+/// Remove a label of the given type for the specified task, returning the old label for the type, if any
+#[inline]
+pub fn remove_label_for_task<T: Clone + Debug + 'static>(task_id: TaskId) -> Option<T> {
+    with_labels_for_task(task_id, |labels| labels.remove())
+}
+
+/// Get the debug name for a task
+#[inline]
+pub fn get_name_for_task(task_id: TaskId) -> Option<TaskName> {
+    get_label_for_task::<TaskName>(task_id)
+}
+
+/// Set the debug name for a task, returning the old name, if any
+#[inline]
+pub fn set_name_for_task(task_id: TaskId, task_name: impl Into<TaskName>) -> Option<TaskName> {
+    set_label_for_task::<TaskName>(task_id, task_name.into())
+}
+
+/// Gets the `TaskId` of the current task, or `None` if there is no current task.
+pub fn get_current_task() -> Option<TaskId> {
+    ExecutionState::with(|s| Some(s.try_current()?.id()))
+}
+
+/// Get the `TaskId` of the current task.  Panics if there is no current task.
+#[inline]
+pub fn me() -> TaskId {
+    get_current_task().unwrap()
+}
+
 /// Sets the `tag` field of the current task.
 /// Returns the `tag` which was there previously.
+#[allow(deprecated)]
```

Comment:
Don't we also want these to be marked as `#[deprecated]`, since they're the public API we're about to remove?

Reply:
Agreed; done.